src/typescript-service.ts

```diff
@@ -352,13 +352,35 @@ export class TypeScriptService {
 				}
 				const offset: number = ts.getPositionOfLineAndCharacter(sourceFile, params.position.line, params.position.character);
 				const info = configuration.getService().getQuickInfoAtPosition(fileName, offset);
+				this.logger.log(info);
 				if (!info) {
 					return { contents: [] };
 				}
-				const contents: MarkedString[] = [{
-					language: 'typescript',
-					value: ts.displayPartsToString(info.displayParts)
-				}];
+				const contents: (MarkedString | string)[] = [];
+				// Add declaration without the kind
+				const declaration = ts.displayPartsToString(info.displayParts).replace(/^\(.+\)\s+/, '');
+				contents.push({ language: 'typescript', value: declaration });
+				// Add kind with modifiers, e.g. "method (private, ststic)", "class (exported)"
+				if (info.kind) {
+					let kind = '**' + info.kind + '**';
```

Comment:
I'd recommend factoring out the computation of `kind` into a separate method for readability's sake.